### PR TITLE
ANL/Battery version 1.0 

### PR DESCRIPTION
**Pull request by wanghy-anl**
Hello Konor,
I have added a package of Lithium-ion batteries in NHES.Systems.EnergyStorage.LithiumIonBattery. It contains three levels of battery models as well as the simulators.
I also included three sets of "gold" folders and the test scripts as depicted in "https://github.com/idaholab/HYBRID/wiki/Developing-Regression-Tests", namely "Test_LithiumIonBattery_Level_0", "Test_LithiumIonBattery_Level_1" and "Test_LithiumIonBattery_Level_2".

Let me know whether it can pass your test. First time doing so.

Thanks,
Haoyu

### CODE REVIEW

File: Models/NHES/Fluid/ClosureModels/HeatTransfer/package.order [hunk 1]
```diff
@@ -2,4 +2,4 @@ Examples
 alpha_Chen_TwoPhase
 alpha_DittusBoelter
 kandlikar_lowP
-Nu_DittusBoelter_Heat_Cool
```
Comment (klfrick2) [inline Models/NHES/Fluid/ClosureModels/HeatTransfer/package.order:5]: This shouldn't automatically add a space.

File: Models/NHES/Systems/EnergyStorage/LithiumIonBattery/BatteryModels.mo [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+    outputPower = -changeRate;
+
+    if capacityChange < 0 then
+      residualCapacity = if (initialCapacity + capacityChange) < minCapacity
+         then minCapacity else (initialCapacity + capacityChange);
+    else
+      residualCapacity = if (initialCapacity + capacityChange) > fullCapacity
+         then fullCapacity else (initialCapacity + capacityChange);
+    end if;
+
```
Comment (klfrick2) [inline Models/NHES/Systems/EnergyStorage/LithiumIonBattery/BatteryModels.mo:157]: We would like to have a variable that describes the Energy storage in the unit that is in units of W-hr. This is normally used in the energy storage fields.

File: Models/NHES/Systems/EnergyStorage/LithiumIonBattery/BatteryModels.mo [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+      ChargeVoltage = 0.0;
+      TerminalPower = 0.0;
+      Chem_Potential_Energy_Rate = 0.0;
+    end if;
+
+    // Update the ResidualCharge value
+    ResidualCharge = InitialCharge + DeltaCharge;
+
+    // Boundary conditions
+
```
Comment (klfrick2) [inline Models/NHES/Systems/EnergyStorage/LithiumIonBattery/BatteryModels.mo:391]: We would like to have a variable that describes the Energy storage in the unit that is in units of W-hr. This is normally used in the energy storage fields.

File: Models/NHES/Systems/EnergyStorage/LithiumIonBattery/BatteryModels.mo [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+    TerminalPower = powerConnection.W;
+    f = powerConnection.f;
+
```
Comment (klfrick2) [inline Models/NHES/Systems/EnergyStorage/LithiumIonBattery/BatteryModels.mo:394]: In the inputs to the models can we put in the total storage size of the unit? E.g. Max Charge is 400MW-hr. Max input rate is 100MWe.

File: Models/NHES/Systems/EnergyStorage/LithiumIonBattery/Data/Data_TEDS.mo [hunk 5]
```diff
@@ -0,0 +1,17 @@
+within NHES.Systems.EnergyStorage.LithiumIonBattery.Data;
+record Data_TEDS
+
```
Comment (klfrick2) [inline Models/NHES/Systems/EnergyStorage/LithiumIonBattery/Data/Data_TEDS.mo]: I don't think we need this file in the system for the Lithium Ion Battery.

File: Models/NHES/Systems/BalanceOfPlant/Turbine/CS_OTSG_TCV_Pressure_TBV_Power_Control.mo [hunk 6]
```diff
@@ -1,148 +0,0 @@
-within NHES.Systems.BalanceOfPlant.Turbine;
```
Comment (klfrick2) [inline Models/NHES/Systems/BalanceOfPlant/Turbine/CS_OTSG_TCV_Pressure_TBV_Power_Control.mo:1]: This file shouldn't be deleted.

**Review by klfrick2 — CHANGES_REQUESTED** [review]
Comments in the files. In general it looks good, main things to address is making sure all the proper files are in the PR and that the battery models are able to express the power input; max storage capacity; max storage rate as W and W-hr's.

File: Models/NHES/Fluid/ClosureModels/HeatTransfer/package.order (hunk 1 above)
Comment (wanghy-anl) [reply Models/NHES/Fluid/ClosureModels/HeatTransfer/package.order:5]: Let's use your corrections. It's weird that I didn't touch this file at all. Maybe this file is altered by the Dymola software?

File: Models/NHES/Systems/BalanceOfPlant/Turbine/CS_OTSG_TCV_Pressure_TBV_Power_Control.mo (hunk 6 above)
Comment (wanghy-anl) [reply Models/NHES/Systems/BalanceOfPlant/Turbine/CS_OTSG_TCV_Pressure_TBV_Power_Control.mo:1]: Let's use your corrections. It's weird that I didn't touch this file at all. Maybe this file is altered by the Dymola software?

File: Models/NHES/Systems/EnergyStorage/LithiumIonBattery/BatteryModels.mo (hunk 2 above)
Comment (wanghy-anl) [reply Models/NHES/Systems/EnergyStorage/LithiumIonBattery/BatteryModels.mo:157]: No problem. I added two parameters "TotalCapacity" and "InitialEnergyStored", which is in units of W-hr.

File: Models/NHES/Systems/EnergyStorage/LithiumIonBattery/BatteryModels.mo (hunk 3 above)
Comment (wanghy-anl) [reply Models/NHES/Systems/EnergyStorage/LithiumIonBattery/BatteryModels.mo:391]: No problem. I added the variable "ResidualEnergy", which is in units of W-hr.

File: Models/NHES/Systems/EnergyStorage/LithiumIonBattery/BatteryModels.mo (hunk 4 above)
Comment (wanghy-anl) [reply Models/NHES/Systems/EnergyStorage/LithiumIonBattery/BatteryModels.mo:394]: No problem. I added three parameters, "TotalCapacity" in W-hr, "MaxChargeRate" in W, and "MaxDischargeRate" in W.

File: Models/NHES/Systems/EnergyStorage/LithiumIonBattery/Data/Data_TEDS.mo (hunk 5 above)
Comment (wanghy-anl) [reply Models/NHES/Systems/EnergyStorage/LithiumIonBattery/Data/Data_TEDS.mo]: To be honest, I don't know the purpose of this data, either... But I will defer to you regarding whether this line should be deleted. I won't do any changes in this file in the latest push.

**Review by klfrick2 — APPROVED** [review]
These look good.

**Pull request merged**